lib/request.js

```diff
@@ -16,25 +16,25 @@
 var accepts = require('accepts');
 var isIP = require('net').isIP;
 var typeis = require('type-is');
-var http = require('http');
 var fresh = require('fresh');
 var parseRange = require('range-parser');
 var parse = require('parseurl');
 var proxyaddr = require('proxy-addr');
+var setPrototypeOf = require('setprototypeof')
 
 /**
  * Request prototype.
  * @public
  */
-
-var req = Object.create(http.IncomingMessage.prototype)
+exports = module.exports = function (proto) {
+  return setPrototypeOf(req, proto)
```

Comment:
Yeah I agree with this including the edit. The use case of multiple servers was *always* bad with express patching things though so I am not sure this makes it *worse*, but the mixin way seems like it would make it better in general. I am +1 for going that route and calling these other comments resolved.